app/Services/PolygonService.php

```diff
@@ -323,4 +347,59 @@ protected function transformAndStorePoints($geojson, $sitePolygonProperties): ar
 
         return $this->createGeojsonModels($polygonsGeojson, $sitePolygonProperties);
     }
+
+    public function isFieldInvalid($field, $value)
+    {
+        if (is_null($value) || $value === '') {
+            return true;
+        }
+
+        switch ($field) {
+            case 'plantstart':
+                return ! $this->isValidDate($value);
+            case 'plantend':
+                return ! $this->isValidDate($value);
+            case 'practice':
+                return ! $this->areValidPractices($value);
+            case 'target_sys':
+                return ! in_array($value, self::VALID_SYSTEMS);
+            case 'distr':
+                return ! $this->areValidDistributions($value);
+            case 'num_trees':
+                return ! filter_var($value, FILTER_VALIDATE_INT);
+            default:
+                return false;
+        }
+    }
+
+    private function areValidPractices($value)
+    {
+        $practices = explode(',', $value);
+        foreach ($practices as $practice) {
+            if (! in_array(trim($practice), self::VALID_PRACTICES)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private function areValidDistributions($value)
+    {
+        $distributions = explode(',', $value);
+        foreach ($distributions as $distribution) {
+            if (! in_array(trim($distribution), self::VALID_DISTRIBUTIONS)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private function isValidDate($date)
+    {
+        $d = DateTime::createFromFormat('Y-m-d', $date);
+
+        return $d && $d->format('Y-m-d') === $date;
+    }
```

Comment:
isn't it possible that the DateTime parse throws an error? can you catch in the flow?

Reply:
try catch added.